src/adr/0035-distributed-listeners-configuration.md

```diff
@@ -0,0 +1,145 @@
+# 35. Distributed Mailbox Listeners Configuration
+
+Date: 2020-04-23
+
+## Status
+
+Proposed
+
+Supercedes [26. Removing a configured additional MailboxListener](0026-removing-configured-additional-mailboxListeners.md)
+
+## Context
+
+James enables a user to register additional mailbox listeners.
+
+The distributed James server is handling mailbox event processing (mailboxListener execution) using a RabbitMQ work-queue
+per listener.
+
+Mailbox listeners can be registered to be triggered every time an event is generated by user interaction with their 
+mailbox. They are being executed in a distributed fashion following the workqueue messaging pattern. The "group" is an 
+attribute of the mailbox listener identifying to which work queue they belong.
+
+Currently, mailbox listeners are determined by the guice bindings of the server and additional mailbox listeners defined
+via configuration files.
+
+While the configuration might be specific for each James server, what actually is defined in RabbitMQ is common. 
+Heterogeneous configuration might then result in unpredictable RabbitMQ resource status. This was left as a limitation
+of [26. Removing a configured additional MailboxListener](0026-removing-configured-additional-mailboxListeners.md).
+
+## Decision
+
+We need to centralize the definition of mailbox listeners.
+
+An event sourcing system will track the configured mailbox listeners.
+
+It will have the following commands:
+
+ - **AddListener**: Add a given listener. This should be rejected if the group is already used.
```

Comment:
What about changeListenerConfiguration?

Reply:
Configuration changes are not supported. The administrator is expected to remove the given listener, then add it again.